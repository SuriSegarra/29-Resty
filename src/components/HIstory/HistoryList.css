.History {
        flex-basis: 0;
        height: calc(100vh - 9em);
        overflow: auto;
        display: block;
        margin-block-start: 1em;
        margin-block-end: 1em;
         margin-inline-start: 0px;
         margin-inline-end: 0px;
         padding-inline-start: 40px;
    li {
        /* border: 1px solid black;
        color: white;
        background-color: #204060; */
        border: 1px groove #ccc;
        color: #eee;
        cursor: pointer;
        display: flex;
        flex-direction: column;
        font-size: .75em;
        margin-bottom: .5em;
        padding: .33em;
        border-radius: 3px;
        background: #2d5986;
        
    }
}

aside {
    margin-right: 1em;
    width: 25%;
}

h2 {
    font-weight: 700;
    margin-bottom: .5em;
    font-family: Verdana,Geneva,Tahoma,sans-serif;
    position: relative;
    left: 4%;
    font-size: 20px;
}
 
