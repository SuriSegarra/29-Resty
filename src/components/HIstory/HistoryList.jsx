import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const HistoryList = ({ history }) => {
  //neither the URL not the method are gonna be unique, you could hit the same URl multiple times. You can use the same method multiple times. thats why 'i'
  const historyElements = history.map((item, i) => (
    <li key={i}>
      <HistoryItem {...item} />
    </li>
  ));

  return (
    <ul>
      {historyElements}
    </ul>
  );
};

HistoryList.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;
