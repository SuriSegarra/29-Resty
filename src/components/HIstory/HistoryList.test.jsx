import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from './HistoryList';

const history = [
  {
   
    url: 'https//:testur.com',
    method: 'POST'
  }
];

describe('HistoryList', () => {
  it('renders Characters', () => {
    const wrapper = shallow(<HistoryList history={history} />);
    expect(wrapper).toMatchSnapshot();
  });
});