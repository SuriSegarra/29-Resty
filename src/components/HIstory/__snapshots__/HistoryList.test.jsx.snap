// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList renders Characters 1`] = `
<ul>
  <li
    key="0"
  >
    <HistoryItem
      method="POST"
      url="https//:testur.com"
    />
  </li>
</ul>
`;
